inference-engine/tests/functional/plugin/shared/src/behavior/caching_tests.cpp

```diff
@@ -0,0 +1,189 @@
+    res.push_back({inputShapeWrapper(ngraph::builder::subgraph::makeConvBias, {1, 3, 24, 24}),
+                   "ConvBias"});
+    res.push_back({inputShapeWrapper(ngraph::builder::subgraph::makeReadConcatSplitAssign, {1, 1, 2, 4}),
+                   "ReadConcatSplitAssign"});
+
+    return res;
+}
+
+bool LoadNetworkCacheTestBase::importExportSupported(InferenceEngine::Core& ie) const {
+    std::vector<std::string> supportedMetricKeys = ie.GetMetric(targetDevice, METRIC_KEY(SUPPORTED_METRICS));
+    auto it = std::find(supportedMetricKeys.begin(), supportedMetricKeys.end(),
+                        METRIC_KEY(IMPORT_EXPORT_SUPPORT));
+    bool supported = (it != supportedMetricKeys.end()) &&
+                     ie.GetMetric(targetDevice, METRIC_KEY(IMPORT_EXPORT_SUPPORT));
+    return supported;
+}
+
+std::string LoadNetworkCacheTestBase::getTestCaseName(testing::TestParamInfo<loadNetworkCacheParams> obj) {
+    auto param = obj.param;
+    auto funcName = std::get<1>(std::get<0>(param));
+    auto precision = std::get<1>(param);
+    auto deviceName = std::get<2>(param);
+    return funcName + "_" + ngraph::element::Type(precision).get_type_name() + "_" + deviceName;
+}
+
+void LoadNetworkCacheTestBase::SetUp() {
+    nGraphFunctionWithName funcPair;
+    std::tie(funcPair, m_precision, targetDevice) = GetParam();
+    auto fGen = std::get<0>(funcPair);
+    m_functionName = std::get<1>(funcPair);
+    try {
+        function = fGen(m_precision);
+    } catch (...) {
+        SKIP();
+    }
+
+    std::stringstream ss;
+    auto hash = std::hash<std::string>()(GetTestName());
+    ss << std::to_string(hash) << "_" << std::this_thread::get_id() << "_" << GetTimestamp();
+    m_cacheFolderName = ss.str();
+}
+
+void LoadNetworkCacheTestBase::TearDown() {
+    CommonTestUtils::removeFilesWithExt(m_cacheFolderName, "blob");
+    std::remove(m_cacheFolderName.c_str());
+}
+
+void LoadNetworkCacheTestBase::Run() {
+    auto compareOutputs = [&](const std::vector<InferenceEngine::Blob::Ptr>& expected,
+                              const std::vector<InferenceEngine::Blob::Ptr>& actual) {
+        ASSERT_EQ(expected.size(), actual.size());
+        for (size_t i = 0; i < expected.size(); i++) {
+            const auto& expPtr = expected[i];
+            const auto& actPtr = actual[i];
+            ASSERT_NO_THROW(Compare(expPtr, actPtr));
+        }
+    };
+    if (!function) {
+        GTEST_COUT << "Can't create function " << m_functionName << " with precision " << m_precision.get_type_name() << std::endl;
+        SKIP();
+    }
+    if (!importExportSupported(*core)) {
+        GTEST_COUT << "Plugin doesn't support import and export - skipping test" << std::endl;
+        SKIP();
+    }
+    cnnNetwork = CNNNetwork{function};
+    ConfigureNetwork();
+    try {
+        core = std::make_shared<Core>();
+        executableNetwork = core->LoadNetwork(cnnNetwork, targetDevice, configuration);
```

Comment:
I suppose we need to destroy executableNetwork and inference request which hold reference to plugins and it leads to fail in the next LoadNetwork for MYRIAD plugin.